lib/fluent/plugin/filter_record_transformer.rb

```diff
@@ -103,6 +107,21 @@ def filter_stream(tag, es)
             time = Fluent::EventTime.from_time(Time.at(new_record[@renew_time_key].to_f))
           end
           @key_deleters.each { |deleter| deleter.delete(new_record) } if @key_deleters
+
+          # add track information
+          if @track_key
+            if new_record.has_key?(@track_key)
+              new_record[@track_key] = "#{new_record[@track_key]},#{@hostname},#{Time.new.to_i}"
+            else
+              new_record[@track_key] = "#{@hostname},#{Time.new.to_i}"
+            end
+          end
+
+          # add record length information
+          if @record_size_key
+            new_record[@record_size_key] = new_record.to_s.size
+          end
+
```

Comment:
First, could you write test for your implementation?

---

>               new_record[@track_key] = "#{new_record[@track_key]},#{@hostname},#{Time.new.to_i}"

`Time.new.to_i` decays nanosecond precision. We should use `Time.now.to_f` here.

>               new_record[@track_key] = "#{@hostname},#{Time.new.to_i}"

ditto.

Reply:
Ok, I got it,   think you.